### PR TITLE
fix: download protected tarballs

### LINKED ISSUE #75: Cannot download tarball after logging in
**Describe the bug**
Access to packages is restricted to authenticated users, which works fine during browsing. However, on clicking "Download tarball" I get an error:
```json
{
  "error": "authorization required to access package my-package"
}
```

**To Reproduce**
1. Login
2. Navigate to a package with access `$authenticated`
3. Click on "Download tarball"

**Expected behavior**
The tarball should be downloaded.

**Configuration File (cat ~/.config/verdaccio/config.yaml)** 
```yaml
packages:
  '**':
    access: $authenticated
    publish: special
    unpublish: special
    # if package is not available locally, proxy requests to 'npmjs' registry
    proxy: npmjs
```

**Additional context**
I checked the network requests and the download does not contain an `Authorization` header.

### PR DESCRIPTION
**Pull request by juanpicado**
**Type:** bug

The following has been addressed in the PR:

*  There is a related issue? yes
*  Unit or Functional tests are included in the PR

**Description:**

- fix issue with webpack and the usage of port 8080, not need it anymore for UI (calls are proxied correctly) https://github.com/verdaccio/ui/pull/110/files#diff-c6c55c02902deda0abd6ef54eb21384eR35
- fix issue where tokens are not being send it to the endpoint (🔴 master **is** broken)
- fix issue downloading tarballs which are protected (after login) #75

The solution was not that easy as I assumed, as you might know, download a tarball requires an Authorization 
header with the token, this is not possible via browser GET, thus, I had to fetch a blob and download it from it adding the right headers.

https://verdaccio.nyc3.digitaloceanspaces.com/gif/Kapture%202019-07-28%20at%2014.11.30.gif

Resolves #75

### CODE REVIEW

**Comment by DanielRuf** [comment]
> this is not possible via browser GET

Can we not use `e.preventDefault()` + `fetch`?

**Comment by juanpicado** [comment]
> > this is not possible via browser GET
> 
> Can we not use `e.preventDefault()` + `fetch`?

Do you mean?
```
 onClick={(e) => {
   e.preventDefault();
  /* eslint-disable @typescript-eslint/no-non-null-assertion */
  actionItem.handler!(link);
}}
```

**Comment by DanielRuf** [comment]
Not sure what the following is

> actionItem.handler!(link)

I mean native `window.fetch` which can set headers like any AJAX request.

File: src/utils/api.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ function handleResponseType(response: Response): Promise<[boolean, Blob | string
     }
 
     // unfortunatelly on download files there is no header available
-    if (response.url.match(/tgz/).length > 0) {
+    if (response.url && response.url.match(/.tgz/) !== null) {
```
Comment (DanielRuf) [inline src/utils/api.ts]: Do they end with this substring?

Comment (juanpicado) [reply src/utils/api.ts]: All the time.

Comment (DanielRuf) [reply src/utils/api.ts]: Why not `endsWith(string)`?

Comment (juanpicado) [reply src/utils/api.ts]: That's better 🙃

File: src/components/ActionBar/ActionBar.tsx [hunk 2]
```diff
@@ -54,16 +73,34 @@ class ActionBar extends Component {
...
-        );
+        const actionItem: Action = ACTIONS[value];
+        if (actionItem.handler) {
+          const fab = (
+            <Tooltip key={key} title={actionItem['title']}>
+              <Fab
+                /* eslint-disable react/jsx-no-bind */
+                onClick={() => {
+                  /* eslint-disable @typescript-eslint/no-non-null-assertion */
+                  actionItem.handler!(link);
```
Comment (juanpicado) [inline src/components/ActionBar/ActionBar.tsx]: There is a check for `handler` above, I don't know why `eslint` does not understand that, so I used the null assertion to bypass the issue.

**Review by sergiohgz — APPROVED** [review]
LGTM

**Pull request merged**